src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs

```diff
@@ -44,11 +43,12 @@ internal partial class SemanticClassificationViewTaggerProvider : AsynchronousVi
 
         [ImportingConstructor]
         public SemanticClassificationViewTaggerProvider(
+            IThreadingContext threadingContext,
```

Comment:
It feels weird that we're applying `[Obsolete]` attributes on importing constructors, unless they're used in tests. Shouldn't we mark them *all* `[Obsolete]` and update tests to construct these via MEF?

Reply:
➡️ The goal is to update all the tests to use the MEF catalog for all MEF parts. I'm in the process of documenting the goals/practices in #28945, which will likely be added to #28967.

Reply:
So these will all eventually have an Obsolete attribute and tests will be updated? Great!

Reply:
➡️ Yes, filed https://github.com/dotnet/roslyn/issues/29182